### PR TITLE
added packages/molenc/molenc.5.0.0

### PR DESCRIPTION
**Pull request by UnixJunkie**
There is a Butina clustering implementation now.
Also, switched to dolog >= 4.0.0

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
There is a missing dependency on `parmap`:
```
#=== ERROR while compiling molenc.5.0.0 =======================================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.09.0 | pinned(https://github.com/UnixJunkie/molenc/archive/v5.0.0.tar.gz)
# path                 ~/.opam/4.09/.opam-switch/build/molenc.5.0.0
# command              ~/.opam/4.09/bin/dune build -p molenc -j 2
# exit-code            1
# env-file             ~/.opam/log/molenc-23-73711b.env
# output-file          ~/.opam/log/molenc-23-73711b.out
### output ###
# File "src/dune", line 15, characters 48-54:
# 15 |   (libraries molenc bst batteries dolog minicli parmap))
#                                                      ^^^^^^
# Error: Library "parmap" not found.
```

File: packages/molenc/molenc.5.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  # ["cp" "bin/molenc_ph4_type_atoms.py" "%{bin}%/molenc_ph4_type_atoms.py"]
+  ["cp" "bin/molenc_common.py" "%{bin}%/molenc_common.py"]
+  ["cp" "bin/molenc_type_atoms.py" "%{bin}%/molenc_type_atoms.py"]
+  ["cp" "bin/molenc_scan.py" "%{bin}%/molenc_scan.py"]
+  ["cp" "bin/molenc.sh" "%{bin}%/molenc.sh"]
+  ["cp" "bin/molenc_scan.sh" "%{bin}%/molenc_scan.sh"]
+]
+depends: [
+  "bst"
+  "dune"
```
Comment (kit-ty-kate) [inline packages/molenc/molenc.5.0.0/opam]: Could you add the `dune-project` file to the repository next time. This makes easier to check which version of dune is required.

Comment (kit-ty-kate) [inline packages/molenc/molenc.5.0.0/opam]:
```suggestion
  "dune" {>= "1.11"}
```
Oops I was looking at the wrong log, discard my previous (deleted) comment.
Anyway, the `dune-project` file in the repository says this package requires at least dune 1.11

**Comment by kit-ty-kate** [comment]
Is there really no way of fixing the issue with `conf-rdkit`?
```
#=== ERROR while compiling conf-rdkit.0.1 =====================================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.09.0 | git+file:///home/opam/opam-repository
# path                 ~/.opam/4.09/.opam-switch/build/conf-rdkit.0.1
# command              /bin/sh -c c++ test.cpp -o test -I/usr/include/rdkit -lRDKitRDGeneral || c++ test.cpp -o test -I/usr/include/rdkit -lRDGeneral
# exit-code            1
# env-file             ~/.opam/log/conf-rdkit-18-58a313.env
# output-file          ~/.opam/log/conf-rdkit-18-58a313.out
### output ###
# test.cpp:2:10: fatal error: rdkit/RDGeneral/versions.h: No such file or directory
#  #include "rdkit/RDGeneral/versions.h"
#           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
# test.cpp:2:10: fatal error: rdkit/RDGeneral/versions.h: No such file or directory
#  #include "rdkit/RDGeneral/versions.h"
#           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
```

**Comment by UnixJunkie** [comment]
At my work, conf-rdkit installs both on Linux and Mac.
The brew recipe for rdkit (for Mac) was updated this morning.

**Comment by mseri** [comment]
```
#=== ERROR while compiling molenc.5.0.0 =======================================#

 context     2.0.5 | linux/x86_64 | ocaml-system.4.09.0 | file:///home/travis/build/ocaml/opam-repository

# path        ~/.opam/ocaml-system/.opam-switch/build/molenc.5.0.0

# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p molenc -j 1

# exit-code   1

# env-file    ~/.opam/log/molenc-21076-865888.env

# output-file ~/.opam/log/molenc-21076-865888.out

### output ###

# Use Stdlib instead.

# [...]

# If you need to stay compatible with OCaml < 4.07, you can use the 

# stdlib-shims library: https://github.com/ocaml/stdlib-shims

#     ocamlopt src/.molenc.objs/native/molenc__Utls.{cmx,o}

# File "src/utls.ml", line 332, characters 13-31:

# 332 |   Array.sort Pervasives.compare xs;

#                    ^^^^^^^^^^^^^^^^^^

# Alert deprecated: module Stdlib.Pervasives

# Use Stdlib instead.

# 

# If you need to stay compatible with OCaml < 4.07, you can use the 

# stdlib-shims library: https://github.com/ocaml/stdlib-shims
```

**Comment by kit-ty-kate** [comment]
@mseri that's the error i commented on above. Sometimes opam doesn't do the different between warnings and a source of errors

**Comment by mseri** [comment]
Oh sorry. I thought it was a new one

**Comment by UnixJunkie** [comment]
I will try to remove the warnings.
Though I am not sure I can correct conf-rdkit for the CI environment.

**Comment by kit-ty-kate** [comment]
> I will try to remove the warnings.

As I said the warnings are not in cause here. Please refer to my first comment.

> Though I am not sure I can correct conf-rdkit for the CI environment.

What do it need? Is it a difference between versions? If it is maybe it is possible to detect that

**Comment by UnixJunkie** [comment]
Could the CI tests be relaunched for this PR?
Because now, parmap~rc11 _will_ install whatever the ocaml version.
I suspect it should work at least on the Mac VM.
Thanks a lot!

**Comment by thomasblanc** [comment]
@UnixJunkie I restarted Camelus. Sorry it said this is your 4th PR (it actually means the search for your PRs returned 4 pages, nice work).

**Comment by kit-ty-kate** [comment]
> I suspect it should work at least on the Mac VM.

On mac it also fails with:
```
#=== ERROR while compiling conf-rdkit.0.1 =====================================#
# context     2.0.5 | macos/x86_64 | ocaml-system.4.08.1 | file:///Users/travis/build/ocaml/opam-repository
# path        ~/.opam/ocaml-system/.opam-switch/build/conf-rdkit.0.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build sh -c c++ test.cpp -o test -I/usr/local/include/rdkit -L/usr/local/lib -lRDKitRDGeneral
# exit-code   1
# env-file    ~/.opam/log/conf-rdkit-69871-74f651.env
# output-file ~/.opam/log/conf-rdkit-69871-74f651.out
### output ###
# test.cpp:2:10: fatal error: 'rdkit/RDGeneral/versions.h' file not found
# #include "rdkit/RDGeneral/versions.h"
#          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 1 error generated.
```

**Comment by UnixJunkie** [comment]
It could install in the Ubuntu CI environment. What a surprise!

**Comment by kit-ty-kate** [comment]
It finally works! @UnixJunkie do you agree with my change?

**Comment by UnixJunkie** [comment]
@kit-ty-kate Yes, why not. I did not know those flags are needed.
Note that in the future, I could make a Python script to check that rdkit is installed
and working properly, instead of compiling a piece of C++.
If you are interested, I can send it soon.

**Comment by kit-ty-kate** [comment]
> If you are interested, I can send it soon.

Do both are installed at the same time and do both need to be installed? If both should be installed and they are not installed at the same time I think that both should be checked.

**Comment by UnixJunkie** [comment]
> > If you are interested, I can send it soon.
> 
> Do both are installed at the same time and do both need to be installed? If both should be installed and they are not installed at the same time I think that both should be checked.

If the python wrappers are working, it is usually enough to check that the whole library (rdkit) has been installed properly.

**Comment by UnixJunkie** [comment]
OK, I'll create a new PR for that.

**Pull request merged**